core/src/main/java/hudson/PluginManager.java

```diff
@@ -1643,26 +1649,73 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
     @Restricted(NoExternalUse.class)
     @RequirePOST public HttpResponse doCheckUpdatesServer() throws IOException {
         Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
-        try {
-            for (UpdateSite site : Jenkins.getInstance().getUpdateCenter().getSites()) {
-                FormValidation v = site.updateDirectlyNow(DownloadService.signatureCheck);
-                if (v.kind != FormValidation.Kind.OK) {
-                    // TODO crude but enough for now
-                    return v;
-                }
+
+        int sleep = SystemProperties.getInteger(PluginManager.class.getName() + ".sleepTimeMilis", 1000);
```

Comment:
I'm not sure about defaulting to 3 retries in case the property is not specified. Maybe we should default to no retry (as of now) and retry only in case the property is specified.

Reply:
I like this approach, @varyvol